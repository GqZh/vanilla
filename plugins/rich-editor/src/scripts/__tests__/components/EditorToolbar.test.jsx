import React from "react";
import { mount, shallow } from "enzyme";
import EditorMenuItem from "../../components/EditorMenuItem";
import EditorToolbar from "../../components/EditorToolbar";
import Quill from "../../Quill";

jest.mock('quill/core');

test("matches snapshot", () => {
    const quill = new Quill();
    const toolbar = shallow(
        <EditorToolbar quill={quill}/>
    );

    expect(toolbar).toMatchSnapshot();
});

test("generates correct number of <EditorMenuItem /> components", () => {
    const menuItems = {
        foo: {},
        bar: {},
        other: {},
        thing: {},
    };
    const quill = new Quill();
    const toolbar = shallow(
        <EditorToolbar quill={quill} menuItems={menuItems}/>
    );

    expect(toolbar.find(EditorMenuItem).length).toBe(4);
});

test("can receive a custom formatter for the menu item click handler.", () => {
    const mockFormatter = jest.fn();

    const menuItems = {
        bold: {
            active: false,
            formatter: mockFormatter,
        },
    };
    const quill = new Quill();
    const toolbar = mount(
        <EditorToolbar quill={quill} menuItems={menuItems}/>
    );

    toolbar.find(".richEditor-button").simulate("click");
    toolbar.find(".richEditor-button").simulate("click");

    expect(mockFormatter.mock.calls.length).toBe(2);
});

describe("update", () => {
    const quill = new Quill();
    const mockGetFormat = jest.fn();
    mockGetFormat
        .mockReturnValueOnce({bold: true})
        .mockReturnValueOnce({});
    quill.getFormat = mockGetFormat;
    const menuItems = {
        bold: {
            active: false,
        },
    };

    const toolbar = shallow(
        <EditorToolbar quill={quill} menuItems={menuItems}/>
    );

    it("updates boolean type", () => {

        /** @var {EditorToolbar} */
        const instance = toolbar.instance();
        instance.update({});
        expect(toolbar.state("bold").active).toBe(true);
        instance.update({});
        expect(toolbar.state("bold").active).toBe(false);
    });
});
