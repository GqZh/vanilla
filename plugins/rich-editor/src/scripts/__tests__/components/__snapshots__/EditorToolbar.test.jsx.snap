// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  aria-label="Inline Level Formatting Menu"
  className="richEditor-menu"
  role="dialog"
>
  <ul
    aria-label="Inline Level Formatting Menu"
    className="richEditor-menuItems MenuItems"
    role="menubar"
  >
    <EditorMenuItem
      clickHandler={[Function]}
      isActive={false}
      key="0"
      propertyName="bold"
    />
    <EditorMenuItem
      clickHandler={[Function]}
      isActive={false}
      key="1"
      propertyName="italic"
    />
    <EditorMenuItem
      clickHandler={[Function]}
      isActive={false}
      key="2"
      propertyName="strike"
    />
    <EditorMenuItem
      clickHandler={[Function]}
      isActive={false}
      key="3"
      propertyName="code"
    />
    <EditorMenuItem
      clickHandler={[Function]}
      isActive={false}
      key="4"
      propertyName="link"
    />
  </ul>
</div>
`;
